src/quick-lint-js/fe/parse-class.cpp

```diff
@@ -403,9 +403,17 @@ void parser::parse_and_visit_class_or_interface_member(
       QLJS_CASE_CONTEXTUAL_KEYWORD:
       case token_type::identifier:
       case token_type::reserved_keyword_with_escape_sequence: {
-        identifier property_name = p->peek().identifier_name();
-        p->skip();
-        parse_and_visit_field_or_method(property_name);
+        auto parse_and_visit = [this](auto p) {
+          identifier property_name = p->peek().identifier_name();
+          p->skip();
+          parse_and_visit_field_or_method(property_name);
+        };
+        if (p->peek().type == token_type::kw_constructor) {
+          constructor_guard g = p->enter_constructor();
+          parse_and_visit(p);
+        } else {
+          parse_and_visit(p);
+        }
```

Comment:
Nit: Why is `p` a parameter? I think it's already accessible through `this`:
```suggestion
        auto parse_and_visit = [this]() {
          identifier property_name = p->peek().identifier_name();
          p->skip();
          parse_and_visit_field_or_method(property_name);
        };
        if (p->peek().type == token_type::kw_constructor) {
          constructor_guard g = p->enter_constructor();
          parse_and_visit();
        } else {
          parse_and_visit();
        }
```

Reply:
Thanks for the tip, I hadn't noticed that